File: app/src/main/java/com/klinker/android/twitter_l/data/roomdb/entities/FavoriteUser.java
package com.klinker.android.twitter_l.data.roomdb.entities;


import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(tableName = "favorite_users",
        indices = {@Index(value = { "user_twitter_id", "account"}, unique = true), @Index(value = "account"), @Index(value = "user_id")},
        foreignKeys = { @ForeignKey( entity = User.class, parentColumns = "id", childColumns = "user_id", onDelete = ForeignKey.CASCADE)})
public class FavoriteUser {

    @PrimaryKey(autoGenerate = true)
    public Long id;

    @ColumnInfo
    public int account;

    @ColumnInfo(name = "user_id")
    public Long userId;

    @ColumnInfo(name = "user_twitter_id")
    public long twitterId;

    public FavoriteUser(twitter4j.User user, int account) {
        this.id = null;
        this.account = account;
        this.userId = null;
        this.twitterId = user.getId();
    }

    public FavoriteUser(long id, int account, long userId, long twitterId) {
        this.id = id;
        this.account = account;
        this.userId = userId;
        this.twitterId = twitterId;
    }

}

File: app/src/main/java/com/klinker/android/twitter_l/data/roomdb/entities/FavoriteUser.kt
package com.klinker.android.twitter_l.data.roomdb.entities


import androidx.room.*

import twitter4j.User as TwitterUser


@Entity(tableName = "favorite_users",
        indices = [Index(value = ["user_id", "account"], unique = true), Index(value = ["account"])],
        foreignKeys = [ForeignKey(entity = UserInteraction::class, parentColumns = ["id"], childColumns = ["user_id"], onDelete = ForeignKey.CASCADE)])
data class FavoriteUser(@PrimaryKey(autoGenerate = true) val id: Long? = null,
                        @ColumnInfo val account: Int = 1,
                        @ColumnInfo(name = "user_id") val userId: Long) {

    constructor(id: Long? = null, account: Int, user: User) : this(id, account, user.userId!!)

}




//@Entity(tableName = "favorite_users",
//        indices = [Index(value = ["user_id", "account"], unique = true), Index(value = ["account"])],
//        foreignKeys = [ForeignKey(entity = User::class, parentColumns = ["id"], childColumns = ["user_id"], onDelete = ForeignKey.CASCADE)])
//class FavoriteUser {
//
//    @PrimaryKey(autoGenerate = true)
//    var id: Long? = null
//
//    @ColumnInfo
//    var account: Int = 0
//
//    @ColumnInfo(name = "user_id")
//    var userId: Long? = null
//
//
//    constructor(user: TwitterUser, account: Int) {
//        this.id = null
//        this.account = account
//        this.userId = user.id
//    }
//
//    constructor(account: Int, userId: Long, id: Long? = null) {
//        this.id = id
//        this.account = account
//        this.userId = userId
//    }
//
//}